pkg/kubelet/kubelet.go

```diff
@@ -1486,7 +1486,14 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	}
 	switch internalFieldPath {
 	case "status.podIP":
-		return pod.Status.PodIP, nil
+		status, _ := kl.statusManager.GetPodStatus(pod.UID)
+		if status.PodIP != "" {
+			return status.PodIP, nil
+		}
+
+		glog.V(4).Infof("Calling out to container runtime for status of pod %v/%v in order to inject pod IP", pod.Namespace, pod.Name)
+		runtimeStatus, _ := kl.containerRuntime.GetPodStatus(pod.UID, pod.Name, pod.Namespace)
```

Comment:
*should* or *will*? You are ignoring the error code here and that's the reason for my comment.